cmd/cli/plugin/feature/activate.go

```diff
@@ -24,22 +30,112 @@ var FeatureActivateCmd = &cobra.Command{
 }
 
 func init() {
-	FeatureActivateCmd.Flags().StringVarP(&featuregate, "featuregate", "f", "tkg-system", "Activate a Feature gated by a particular FeatureGate")
+	FeatureActivateCmd.Flags().BoolVar(&userAllowsVoidingWarranty, "permanentlyVoidAllSupportGuarantees", false, "Allow for the permanent voiding of all support guarantees for this environment. For some features, e.g. experimental features, if a user sets the activation status to one that does not match the default activation, all support guarantees for this environment will be permanently voided.")
 }
 
 func featureActivate(cmd *cobra.Command, args []string) error {
 	featureName := args[0]
-	featureGateClient, err := featuregateclient.NewFeatureGateClient()
+
+	fgClient, err := featuregateclient.NewFeatureGateClient()
 	if err != nil {
-		return fmt.Errorf("couldn't get featureGateRunner: %w", err)
+		return fmt.Errorf("could not get FeatureGate client: %w", err)
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
 	defer cancel()
 
-	if err := featureGateClient.ActivateFeature(ctx, featureName, featuregate); err != nil {
-		return fmt.Errorf("couldn't activate feature %s: %w", featureName, err)
+	feature, err := fgClient.GetFeature(ctx, featureName)
+	if err != nil {
+		return fmt.Errorf("could not get Feature %s: %w", featureName, err)
+	}
+
+	gates, err := fgClient.GetFeatureGateList(ctx)
+	if err != nil {
+		return fmt.Errorf("could not get FeatureGate List: %w", err)
+	}
+
+	gateName, featRef := featuregateclient.FeatureRefFromGateList(gates, featureName)
+
+	var proceedWithVoidingWarranty bool
+	if willWarrantyBeVoided(featRef, feature) {
+		// The warranty will be voided with the request, so check that user allows it.
+		var userAllows *bool
+		if cmd.Flags().Changed("permanentlyVoidAllSupportGuarantees") {
+			userAllows = &userAllowsVoidingWarranty
+		}
+		proceedWithVoidingWarranty, err = userGivesPermissionToVoidWarranty(feature, userAllows)
+		if err != nil {
+			return fmt.Errorf("could not get user permission to void warranty for Feature %s: %w", featureName, err)
+		}
 	}
-	cmd.Printf("Feature %s Activated", featureName)
+
+	err = fgClient.ActivateFeature(ctx, featureName, proceedWithVoidingWarranty)
+	if err != nil {
+		return fmt.Errorf("could not activate Feature %s gated by FeatureGate %s: %w", featureName, gateName, err)
+	}
+
+	displayActivationWarnings(feature)
```

Comment:
can we display this message before activating?

Reply:
When I did manual testing, having this message before activation did not work out, but after it did, worked great. The reason is because the line that does the activation might error out and cause the problem of having conflicting messages. As Jeff said, "Is that a warning or an error?" 

I think you should either have a warning or an error and not both and by having the activation warning after the activation ensures that the warning only shows up if there's no err. Here's what it looks like currently (shown in [a Gist)](https://gist.github.com/codegold79/0b4e76d437e58173dc17fbb2047e2e69),

```
$ ./build/feature activate small-cache
Warning: Technical preview features are not ready, but are not believed to be dangerous. The feature itself is unsupported, but activating technical preview features does not affect the support status of the environment. Use at your own risk.
Feature small-cache gated by FeatureGate tkg-system-b is activated.
```

Reply:
Per @yharish991's request, I have added a blank line in between the warning and Feature activation notice. It now looks like
```
$ ./build/feature activate small-cache
Warning: Technical preview features are not ready, but are not believed to be dangerous. The feature itself is unsupported, but activating technical preview features does not affect the support status of the environment. Use at your own risk.

Feature small-cache gated by FeatureGate tkg-system-b is activated.
```

Reply:
I would like to display the warning message first and any errors later for a consistent user experience. kubectl does the same, for example, when you try to get a resource that doesn't exist of an API type that is deprecated, it first displays the warning message and error later.

Also, as a user when the warning message is displayed first, it gives me an idea if i want to further perform the operation if any error. Otherwise, if i see the error first, i would figure out how to fix it the error first to perform the operation and then later know that there is a warning when performing the operation. If i had known about the warning earlier, i would have changed my mind about not performing the operation at all.

Reply:
this is a non-blocker though

Reply:
Since it's a non-blocker, I'm going to resolve this comment and the warning/error message issue can be addressed another time.